### PR TITLE
Add new contracts from EthNY

### LINKED ISSUE #198: Use Waffle to test contracts
**Is your feature request related to a problem? Please describe.**
Our smart contracts use a custom testing framework & vyper. This is not great considering we want to use solidity & reduce the maintenance requirements of our project. 

**Describe the solution you'd like**
Move contracts to Solidity and use waffle framework for testing.

### PR DESCRIPTION
**Pull request by karlfloersch**
- [x] I have read and understood the [Plasma Group Contributing Guide and Code of Conduct](https://github.com/plasma-group/pigi/blob/master/.github/CONTRIBUTING.md) and am following those guidelines in this pull request.

------------------------------------------------------------------------

Fixes #198

## Description
This pull request revamps the `contracts` package to use Waffle & also integrates the contract @alexanderattar wrote during the EthNY hackathon. 

The only regret is this PR also squashes some bugs in `core`. However, arguably this is not a problem considering core's functionality is already quite out of date & needs to be updated.

### CODE REVIEW

**Comment by karlfloersch** [comment]
@ben-chain I just edited the PR so that instead of squashing the bugs in core, I just disabled the build & test script as I did for PR https://github.com/plasma-group/pigi/pull/201

File: packages/contracts/contracts/BasicTokenMock.sol [hunk 1]
```diff
@@ -0,0 +1,11 @@
+pragma solidity ^0.5.1;
```
Comment (kfichter) [inline packages/contracts/contracts/BasicTokenMock.sol]: What's this contract for?

File: packages/contracts/contracts/Commitment.sol [hunk 2]
```diff
@@ -0,0 +1,24 @@
+pragma solidity ^0.5.0;
+pragma experimental ABIEncoderV2;
+
+import "openzeppelin-solidity/contracts/math/Math.sol";
```
Comment (kfichter) [inline packages/contracts/contracts/Commitment.sol]: Is this import used?

File: packages/contracts/contracts/BasicTokenMock.sol (hunk 1 above)
Comment (karlfloersch) [reply packages/contracts/contracts/BasicTokenMock.sol]: Ah am using this contract as the ERC20 the deposit contract.

File: packages/contracts/contracts/Commitment.sol (hunk 2 above)
Comment (karlfloersch) [reply packages/contracts/contracts/Commitment.sol]: Nope! Got rid of it because can't think of when it would be.

Comment (alexanderattar) [reply packages/contracts/contracts/Commitment.sol]: I actually don't think the Commitment.sol contract is necessary anymore. Initially we were adding the verifyInclusion method on this and use it within Deposit.sol, but ended up just putting the method directly on the Deposit contract.

Comment (ben-chain) [reply packages/contracts/contracts/Commitment.sol]: Yep--but that was more for hackathon purposes, we would have needed to pull into an external contract eventually :)

Comment (alexanderattar) [reply packages/contracts/contracts/Commitment.sol]: Yeah that's a good point. Probably just better to keep and iterate on it then.

**Pull request merged**